src/simulation/simulation.jl

```diff
@@ -499,15 +583,38 @@ function _apply_warm_start!(model::DecisionModel)
     return
 end
 
+function _update_simulation_state!(sim::Simulation, model::EmulationModel)
```

Comment:
@daniel-thom this function isn't fully implemented yet, there are some design choices to be resolved in the follow up PR.